etna/transforms/timestamp/holiday.py

```diff
@@ -84,13 +138,41 @@ def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
         :
             pd.DataFrame with added holidays
         """
-        if (df.index[1] - df.index[0]) > datetime.timedelta(days=1):
-            raise ValueError("Frequency of data should be no more than daily.")
+        if self.freq == "":
+            raise ValueError("You should specify frequency in your dataset")
```

Comment:
We can't create `TSDataset` without specifying freq, so this check is redundant.